### PR TITLE
# Added BinarySearch algorithm

### PR DESCRIPTION
**Pull request by Huzaib**
## :+1:

### CODE REVIEW

File: divideconquer/BinarySearch.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import java.io.*;
+import java.util.*;
+
+public class BinarySearch {
+
+    static int binarySearch(int[] a, int x, int left, int right) {
```
Comment (rbshealy) [inline divideconquer/BinarySearch.java:6]: use `public static int binarySearch(...)`

File: divideconquer/BinarySearch.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        int mid = (right-left)/2;
+        if(left<right)
```
Comment (rbshealy) [inline divideconquer/BinarySearch.java:8]: Add whitespace to make your code more readable
`int mid = (right-left)/2;`
` `
`if(left<right)`

File: divideconquer/BinarySearch.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        {
+        if(a[mid] ==x)
```
Comment (rbshealy) [inline divideconquer/BinarySearch.java:10]: reformat bracketing to one line
`if(left < right) {`

File: divideconquer/BinarySearch.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        else if(a[mid]>x)
+        return binarySearch(a, x, left, mid);
+        else if(a[mid]<x)
+        return binarySearch(a, x, mid+1, right);
+        else
+        return -1;}
+        else if(left ==right)
+        {if(a[mid] ==x)
+        return mid;}
+	return -1;
```
Comment (rbshealy) [inline divideconquer/BinarySearch.java:21]: Please use indentation and brackets, and clean up these if else-if statements
This is hard to read

File: divideconquer/BinarySearch.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    public static void main(String[] args) {
+        FastScanner scanner = new FastScanner(System.in);
+        int n = scanner.nextInt();
+        int[] a = new int[n];
+        for (int i = 0; i < n; i++) {
```
Comment (rbshealy) [inline divideconquer/BinarySearch.java:28]: add white space between declaration/assignment and logic
i.e.
`int[] a = new int[n];`
` `
`for (int i = 0; i < n; i++) {`

Comment (Huzaib) [reply divideconquer/BinarySearch.java:28]: Ok I am on it

**Comment by alphagroupsnsw** [comment]
Sent from Yahoo Mail for iPhone


On Thursday, August 6, 2020, 3:36 am, Huzaib <notifications@github.com> wrote:




@Huzaib commented on this pull request.

In divideconquer/BinarySearch.java:
> +        int[] a = new int[n];
+        for (int i = 0; i < n; i++) {

Ok I am on it

**Comment by Huzaib** [comment]
Sir, will you look at it now?

**Comment by rbshealy** [comment]
> Sir, will you look at it now?

Oddly enough I can't see the changes you've made

**Comment by Huzaib** [comment]
Sorry sir, it was my mistake 
I had uploaded the same file again
Sorry for the confusion

**Comment by rbshealy** [comment]
Resolve all issues and then Ill take another look

**Comment by Huzaib** [comment]
I think everything is done
Can you give it a look please?

File: divideconquer/BinarySearch.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+public class BinarySearch {
+
+    public static int binarySearch(int[] a, int x, int left, int right) {
+	    
+	    int mid = low + (high-low)/2;
+	    
+	    if(left<=right) {
+		    
+		    if (arr[mid] == u) {
+
```
Comment (afadhitya) [inline divideconquer/BinarySearch.java:14]: Please clear unnecessary whitespace

File: divideconquer/BinarySearch.java [hunk 7]
```diff
@@ -0,0 +1,97 @@
+import java.io.*;
+
+import java.util.*;
+
+public class BinarySearch {
+
+    public static int binarySearch(int[] a, int x, int left, int right) {
```
Comment (afadhitya) [inline divideconquer/BinarySearch.java:7]: I think it will be better if you add javadoc too

File: divideconquer/BinarySearch.java [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+          b[i] = scanner.nextInt();
+        }
+	    
+        for (int i = 0; i < m; i++) {
+		
+            System.out.print(binarySearch(a, b[i], 0, a.length-1) + " ");
+        }
+    }
+	
+    static class FastScanner {
```
Comment (afadhitya) [inline divideconquer/BinarySearch.java:60]: Please add the access modifier

Comment (afadhitya) [reply divideconquer/BinarySearch.java:60]: Also in other field and method
